### PR TITLE
Revise 5.7.11.1.8 Color glyphs as a graph

### PR DESCRIPTION
**Pull request by PeterConstable**

### CODE REVIEW

**Review by drott — COMMENTED** [review]
As this continues along the lines of yours and @rsheeter 's discussion on layering, I'd defer the review of this to Rod, if you don't mind.

File: colr-gradients-spec.md [hunk 1]
```diff
@@ -1228,29 +1223,79 @@ not require the corresponding graph of paint tables to be re-processed. As a
...
-cycle is formed. Pseudocode:
+**5.7.11.1.8 Color glyphs as a graph**
+
+When using version 1 formats, a color glyph is defined by a directed, acyclic
+graph of linked paint tables. For each BaseGlyphV1Record, the paint table
+referenced by that record is the root of a graph defining a color glyph
+composition.
+
+A graph is comprised of the maximal set of paint tables linked by parent/child
+(subtable) linkages. The links can be direct, in the case of paint table formats
```
Comment (rsheeter) [inline colr-gradients-spec.md:1234]: Maximal throws me a bit here when reading, as does the description of the maximal set of parent/child links followed by discussion of indirect links that seem like they aren't what one would typically think of as parent/child links.

I suggest language more akin to description of a closure: the set of paints reachable from the root. For example:

The graph for a given color glyph is made up of all paint tables reachable from the BaseGlyphV1Record. The BaseGlyphV1Record and several paint table formats use direct links, that is, they include a forward offset to a paint subtable. Two paint formats make indirect links:

...discussion of PaintColrLayers and PaintColrGlyph as you have it...

File: colr-gradients-spec.md [hunk 2]
```diff
@@ -1228,29 +1223,79 @@ not require the corresponding graph of paint tables to be re-processed. As a
...
+The graph for a color glyph is a combination of paint table using any of the
+paint table formats. The simplest color glyph definition would consist of a
+PaintGlyph table linked to a basic fill table (PaintSolid, PaintLinearGradient
+or PaintRadialGradient). But the graph can be arbitrarily complex, with an
+arbitrary depth of paint nodes (to the limits inherent in the formats).
+
+The graph can define a visual element in a single layer, or many elements in
+many layers. The concept of layers, as distinct visual elements stacked in a
+z-order, is not precisely defined in relation to the complexity of the graph.
+Each separate visual element requires a leaf node, but leaf nodes can be re-used
```
Comment (rsheeter) [inline colr-gradients-spec.md:1257]: Nit: 5.7.11.1.7 covers reuse of more than just leaves. Perhaps change to ", but nodes can be re-used"

File: colr-gradients-spec.md [hunk 3]
```diff
@@ -1228,29 +1223,79 @@ not require the corresponding graph of paint tables to be re-processed. As a
...
+
+![A PaintLinearGradient used as a compositing mask is a leaf node in the graph but does not add a distinct visual element.](images/colr_gradient_mask.png)
+
+**Figure 5.34 Graph with a leaf node that isn't a distinct visual element.**
+
+Thus, the generalization that can be made regarding the relationship between the
+number of layers and the nature of the graph is that the number of distinct
+root-to-leaf paths will be greater than or equal to the number of layers.
+
+There is one constraint imposed on the complexity of the graph: the graph shall
```
Comment (rsheeter) [inline colr-gradients-spec.md:1276]: (Optional) Might be worth adding a new section header, say `**5.7.11.1.9 Graph shall be acyclic**, and linking directly to that from places that say things like "See 5.7.11.1.8 regarding requirements that the graph be acyclic."

**Review by rsheeter — APPROVED** [review]
Generally LGTM, left a few small comments

**Pull request merged**